packages/flutter/lib/src/animation/tween.dart

```diff
@@ -0,0 +1,376 @@
+  /// Stop calling the listener every time the progress of the performance changes.
+  void removeListener(VoidCallback listener);
+  /// Calls listener every time the status of the performance changes.
+  void addStatusListener(PerformanceStatusListener listener);
+  /// Stops calling the listener every time the status of the performance changes.
+  void removeStatusListener(PerformanceStatusListener listener);
+
+  /// The current status of this animation.
+  PerformanceStatus get status;
+
+  /// The current direction of the animation.
+  AnimationDirection get direction;
+
+  /// The direction used to select the current curve.
+  ///
+  /// The curve direction is only reset when we hit the beginning or the end of
+  /// the timeline to avoid discontinuities in the value of any variables this
+  /// performance is used to animate.
+  AnimationDirection get curveDirection;
+
+  /// Whether this animation is stopped at the beginning.
+  bool get isDismissed => status == PerformanceStatus.dismissed;
+
+  /// Whether this animation is stopped at the end.
+  bool get isCompleted => status == PerformanceStatus.completed;
+
+  String toString() {
+    return '$runtimeType(${toStringDetails()})';
+  }
+  String toStringDetails() {
+    assert(status != null);
+    assert(direction != null);
+    String icon;
+    switch (status) {
+      case PerformanceStatus.forward:
+        icon = '\u25B6'; // >
+        break;
+      case PerformanceStatus.reverse:
+        icon = '\u25C0'; // <
+        break;
+      case PerformanceStatus.completed:
+        switch (direction) {
+          case AnimationDirection.forward:
+            icon = '\u23ED'; // >>|
+            break;
+          case AnimationDirection.reverse:
+            icon = '\u29CF'; // <|
+            break;
+        }
+        break;
+      case PerformanceStatus.dismissed:
+        switch (direction) {
+          case AnimationDirection.forward:
+            icon = '\u29D0'; // |>
+            break;
+          case AnimationDirection.reverse:
+            icon = '\u23EE'; // |<<
+            break;
+        }
+        break;
+    }
+    assert(icon != null);
+    return '$icon';
+  }
+}
+
+abstract class Evaluatable<T> {
+  const Evaluatable();
+
+  T evaluate(double t);
+
+  WatchableValue<T> watch(Animation parent) {
+    return new WatchableValue<T>(parent: parent, evaluatable: this);
+  }
+}
+
+class WatchableValue<T> extends Watchable with ProxyWatchableMixin {
+  WatchableValue({ this.parent, this.evaluatable });
+
+  final Animation parent;
+  final Evaluatable<T> evaluatable;
+
+  T get value => evaluatable.evaluate(parent.progress);
+}
+
+abstract class Animation extends Watchable {
+  /// The current progress of this animation (a value from 0.0 to 1.0).
+  double get progress;
+
+  String toStringDetails() {
+    return '${super.toStringDetails()} ${progress.toStringAsFixed(3)}';
+  }
+}
+
+abstract class ProxyWatchableMixin implements Watchable {
```

Comment:
move this guy to just below Watchable since he depends only on Watchable and is depended on by people above like WatchableValue